docs/DL-2024-05-20-key-authority-declaration.md

```diff
@@ -0,0 +1,82 @@
+
+DL (May 20, 2024)
+
+# Key Authority Declaration
+
+## Context
+
+Due to the [Change rule circumvention issue](https://github.com/holochain/deepkey/issues/23), we
+must commit additional information on the app chains to designate which change rules are the
+trusted authority for managing a key.
+
+
+## Options
+
+### 1. `key_authority: AgentPubKey`
+
+Identifies the deepkey agent who is the authority for key updates.
+
+This is used to determine which change rules must be followed for updating the keys.
+
+#### Validation process when claiming an app agent
+
+- Check that the `KeyAnchor` in deepkey has a create made by the `key_authority` agent.
+
+#### Validation process when updating an app agent
+
+- Lookup the previous agent record to get the `key_authority`
+- Check that the `KeyAnchor` in deepkey has a create made by the `key_authority` agent.
+- Check that the valid create is an update to the previous key.
+
+#### Questions
+
+#####  A. Does deepkey's validation guarantee that the existence of the create means it is a valid udpate?
+
+Deepkey's validation will guarantee that the anchor create was preceeded by a valid registration
+(ie. the registration had all the valid signatures).
+
+#####  B. What happens when the malicious deepkey tries to update the key?
+
+The validation will fail because the malicious deepkey agent does not batch the `key_authority`
+
+
+### 2. `key_authority: ActionHash`
+
+Points to the KSR authority.
+
+#### Validation process when claiming an app agent
+
+- Check that the `KeyAnchor` in deepkey has a create made by an agent that is a member of the
+  `key_authority` (KSR)
```

Comment:
Is the idea here that we could be checking for authorship by multiple agents, hence allowing new registrations by multiple agents? Is there an advantage to this?

Reply:
Yes, this follows the original design of deepkey to have multiple devices all managing the same keyset.  The only part about this solution that seems problematic to me is how it changes the Holochain integration spec from being a generic `AgentPubKey` comparison to specific app knowledge about the KSR and invite acceptance entry structs.  It is possible to lookup the KSR using the `AgentPubKey` which means option 1 could evolve to option 2 without changing the `key_authority` hash type.